### DISCUSSION TITLE
Random Sort - per day

### DISCUSSION THREAD

**Discussion by nils-a** [microsoft-search/pnp-modern-search#3509] [Q&A] [answered]
### Goal:

I'd like to highlight one item of a list, randomly per day.
Something akin to a "quote of the day" with a list of quotes and a WebPart that shows the one "quote of the day". Randomly, a different one, each day.

Theoretically, this could work, by using the search and setting the sort to `[random:seed={CurrentDate}]`. This would modify the `seed` each day, and the sort will be relatively stable for the same seeds. (Docs say re-index might change the sort for the same seed.)

However, the syntax above does not work. The json sent to the search is this: 

```json
{
  //... many other stuff ...
"SortList":{
 "results":[{
   "Property":"[random:seed={CurrentDate}]",
   "Direction":0
  }]}
  //... many other stuff ...
}
```

and the result is an error code 400:

> "We didn't understand your search terms. Make sure they're using proper syntax."

Now, manually setting the sort to `[random:seed=7]` or any other seed does exactly what I want. So I'm guessing it is "only" the replacement of the token `{CurrentDate}` that does not work here.

### Question:

Would it be possible to enable the use of the tokens in the sort settings, as well as the query?

**Comment by nils-a** [comment]
As a workaround, I solved this in the display-template.

I used the following snippet:

```handlebars
<div class="template--custom">
  {{#with (itemAt data.items (random 0 @root.data.totalItemsCount) ) as |x|}}
     <a href="{{x.Path}}">{{x.Title}}</a>
  {{/with}}
</div>
```

**Comment by wobba** [comment, marked as answer]
I added a work item to support an integer token to be used in the sortlist formula. Should be fairly simple. Using {CurrentDate} will work once we support token parsing in that field. And I'll add another token as well.
